reflex-dom-core/src/Reflex/Dom/Builder/Class.hs

```diff
@@ -150,11 +150,41 @@ class (Monad m, Reflex t, DomSpace (DomBuilderSpace m), MonadAdjust t m) => DomB
                    => m ()
   notReady = lift notReady
 
+{- 2017-05-19 dridus: Commented out per Ryan Trinkle. Note that this does not support mount state and possibly has issues with unready child handling.
+
 class DomBuilder t m => MountableDomBuilder t m where
   type DomFragment m :: *
   buildDomFragment :: m a -> m (DomFragment m, a)
   mountDomFragment :: DomFragment m -> Event t (DomFragment m) -> m ()
 
+-}
+
+-- |'MonadMountStatus' represents an action that can be aware of whether the corresponding DOM built by the action is present within the document yet or not.
+-- Its primary use is to integrate with external libraries which need to be invoked only when DOM structures are installed in the document.
+--
+-- ___Note:___ once the current scope is replaced, any 'performEvent's in the scope will be cancelled and so if you want to observe the 'Unmounted' status
+-- you have to plumb the mount state dynamic back out of the current scope so the parent scope can react to it.
+class Monad m => MonadMountStatus t m | m -> t where
```

Comment:
Let's see if we can avoid using the word "Monad" in the name.  I've  been trying to move away from that (MonadAdjust, etc., ought to be renamed as well, I think, but that's for another time).